pkg/scheduler/ugm/manager.go

```diff
@@ -136,6 +136,11 @@ func (m *Manager) DecreaseTrackedResource(queuePath, applicationID string, usage
 	if !decreased {
 		return decreased
 	}
+
+	group := m.ensureGroup(user, queuePath)
+	groupTracker := m.getGroupTracker(group, false)
```

Comment:
or May be calling `userTracker.getGroupForApp(applicationID)` is more appropriate and simple enough? If there is no mapping exists or group is empty, then definitely something is wrong and it is an error. We should log and move on.

Reply:
Yeah, it's good to use `userTracker.getGroupForApp(applicationID)` after we get the user tracker. We can't use it after we call `userTracker.decreaseTrackedResource`, because the `removeApp` will remove the group tracker in the user tracker.

https://github.com/apache/yunikorn-core/blob/e506b13dea7a5b0621c467cff3c8e36812454846/pkg/scheduler/ugm/user_tracker.go#L81-L88